File: TiaCodegen/Commands/S_Move.cs
using TiaCodegen.Interfaces;

namespace TiaCodegen.Commands
{
    public class S_Move : BaseOperationOrSignal, IFunctionOperation
    {
        public S_Move(params IOperationOrSignal[] operationOrSignals)
            : base(operationOrSignals)
        { }

        public override int Cardinality { get { return 1; } }
    }
}

File: TiaCodegen-ts/src/Commands/S_Move.ts
import { IOperationOrSignal } from '../Interfaces/IOperationOrSignal';
import { IFunctionOperation } from '../Interfaces/IFunctionOperation';
import { BaseOperationOrSignal } from './BaseOperationOrSignal';

export class S_Move extends BaseOperationOrSignal implements IFunctionOperation {
    constructor(...operationOrSignals: IOperationOrSignal[]) {
        super(...operationOrSignals);
    }

    get cardinality(): number {
        return 1;
    }
    set cardinality(_value: number) {
        // readonly, ignore
    }
}
